lib/marqeta/api_error.rb

```diff
@@ -0,0 +1,35 @@
+module Marqeta
+  class ApiError < StandardError
+    def initialize(response)
+      @response = response
+    end
+
+    def http_code
+      response.code
+    end
+
+    def http_code_description
+      RestClient::STATUSES[http_code]
+    end
+
+    def error_code
+      response_hash['error_code']
+    end
+
+    def error_message
+      response_hash['error_message']
+    end
+
+    def to_s
+      "#{http_code} #{http_code_description}: #{error_code} #{error_message}"
+    end
+
+    private
+
+    attr_reader :response
+
+    def response_hash
+      @response_hash ||= JSON.parse(@response)
```

Comment:
What if the `@response` is unparseable? Might be worth rescuing this.

Reply:
Hmm what do you suggest I do in the rescue block? Just return an empty hash?

Reply:
Perhaps `to_s` should behave differently depending on the whether it can parse the response or not? e.g.

```
  def to_s
    if parsable_response?
      "#{http_code} #{http_code_description}: #{error_code} #{error_message}"
    else
      "#{http_code} #{http_code_description}: #{raw_response}"
    end
  end
```

Reply:
But then what to do with the methods I'm expecting to be able to call from the Financeit code? (`error_code` and `error_message`).

Reply:
Oh how about I'll make a default response_hash